src/Microsoft.Health.Fhir.SqlServer/Features/ChangeFeed/SqlServerFhirResourceChangeDataStore.cs

```diff
@@ -0,0 +1,111 @@
+        /// <exception cref="Microsoft.Data.SqlClient.SqlException">Thrown when SQL Server returns a warning or error.</exception>
+        /// <exception cref="System.TimeoutException">Thrown when the time allotted for a process or operation has expired.</exception>
+        /// <exception cref="System.Exception">Thrown when errors occur during execution.</exception>
+        public async Task<IReadOnlyCollection<IResourceChangeData>> GetRecordsAsync(long startId, int pageSize, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsGte(startId, 0, nameof(startId));
+            EnsureArg.IsGte(pageSize, 0, nameof(pageSize));
+
+            var listResourceChangeData = new List<ResourceChangeData>();
+            try
+            {
+                using (SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync(cancellationToken: cancellationToken))
+                {
+                    await sqlConnection.OpenAsync(cancellationToken);
+                    using (SqlCommand sqlCommand = new SqlCommand("dbo.FetchResourceChanges", sqlConnection))
+                    {
+                        sqlCommand.CommandType = CommandType.StoredProcedure;
+                        sqlCommand.Parameters.AddWithValue("@startId", SqlDbType.BigInt).Value = startId;
+                        sqlCommand.Parameters.AddWithValue("@pageSize", SqlDbType.Int).Value = pageSize;
+                        using (SqlDataReader sqlDataReader = await sqlCommand.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                        {
+                            while (await sqlDataReader.ReadAsync(cancellationToken))
+                            {
+                                listResourceChangeData.Add(new ResourceChangeData
+                                {
+                                    Id = (long)sqlDataReader["Id"],
+                                    Timestamp = DateTime.SpecifyKind((DateTime)sqlDataReader["Timestamp"], DateTimeKind.Utc),
+                                    ResourceId = (string)sqlDataReader["ResourceId"],
+                                    ResourceTypeId = (short)sqlDataReader["ResourceTypeId"],
+                                    ResourceVersion = (int)sqlDataReader["ResourceVersion"],
+                                    ResourceChangeTypeId = (byte)sqlDataReader["ResourceChangeTypeId"],
+                                });
+                            }
+                        }
+
+                        return listResourceChangeData;
+                    }
+                }
+            }
+            catch (SqlException ex)
+            {
+                switch (ex.Number)
+                {
+                    case TIMEOUTEXPIRED:
+                        throw new TimeoutException(ex.Message, ex);
+                    default:
+                        _logger.LogError(ex, Resources.SqlExceptionOccurredWhenFetchingResourceChanges);
```

Comment:
>LogError [](http://example.com/codeflow?start=32&length=8)

would it log `ex.Number`? if not, I would suggest to add it, just make it easier on person who would need to expand this switch.

Reply:
Added "ex.Number" for logging.